metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -494,6 +492,10 @@ def run(self) -> None:
                 self.final_status = "cancelled"
                 logger.error("Caught error", exc_info=e)
                 raise
+            except Exception as exc:
+                self.final_status = "pipeline_failure"
+                logger.error("pipline run error: ", exc_info=exc)
```

Comment:
I would've thought this log line would be redundant, since we log for any exception as part of entrypoints.py

Can you provide more details about this?
> with cut off log in report.

Reply:
Ingestion Logs are put into report by self._notify_reporters_on_ingestion_completion() method in finally clause.
So if we don't log pipeline exception before finally cause code block is called, exception that caused pipeline failure is not present in the report.

It can be tested by mangling host_port in any recipe and checking a report in ingestion page.

Reply:
What do you think about handling redundancy?

Should I wrap this exception up PipelineRunError and handle it in entrypoints.py without calling logger.exception(f"Command failed: {exc}") ?
I don't like import from pipeline to entrypoints, because there is intermediary module ingest_cli.

Another option is to cut tracebacks:
```
except Exception as exc:
   self.final_status = "pipeline_failure"
   logger.error("pipline run error: ", exc_info=exc.with_traceback(None))
   raise exc from None```